packages/firestore/src/local/memory_mutation_queue.ts

```diff
@@ -89,13 +91,13 @@ export class MemoryMutationQueue implements MutationQueue {
   }
 
   getLastStreamToken(
-    transaction: PersistenceTransaction
+    _transaction: PersistenceTransaction
   ): PersistencePromise<ProtoByteString> {
     return PersistencePromise.resolve(this.lastStreamToken);
   }
 
   setLastStreamToken(
-    transaction: PersistenceTransaction,
+    _transaction: PersistenceTransaction,
```

Comment:
Can we enable `after-used`? (see https://eslint.org/docs/rules/no-unused-vars#args). That would also make the Storage PR smaller, and the code prettier :)

Reply:
I'd like to have `_` prefixed to all unused parameters regardless of the position, because it's a nice visual cue to show it is unused, which improves the code readability.
@mikelehen Would you like to help us break the tie here? :)

Reply:
I'm not in favor of `after-used` since I do think it would end up with our code being inconsistent in a weird way (some unused parameters have `_` and others don't, depending on the order).  I want the code to be consistent.

That said, I'm on the fence of which way I want to be consistent (having underscores or no underscores for unused parameters).  To be honest, scanning through the diff, I'm not sure how much it's adding.  The vast majority of the violations are cases where we're just implementing a contract (an interface method or callback function) which leads to intentional "unused" parameters.

The `_` makes it more visible that they're unused but VS Code actually colors unused variables differently already, so I'm not sure how much value the `_` adds.

I did notice that we had one case where the unused parameter caught by eslint _was_ actually indicative of a coding mistake, though you missed it in your fixing. 😛 https://github.com/firebase/firebase-js-sdk/pull/1863

@schmidt-sebastian and @wilhuff have both gently pushed back on this (due to the noise and overloading how we use `_` too much) and mostly the lint error seems to be triggering false-positives, and so I'm worried it may just be adding unnecessary visual noise to our code.

So at this point, I think I'm actually leaning towards disabling `no-unused-vars` for parameters, at least in Firestore...  Sorry.   WDYT @Feiyang1

Reply:
I think there is value for `_` even for intentional "unused" variables, that is `_` makes the intention explicit. 
`_` is also helpful during code review since code review tools won't color unused variables differently like VSCode. Similarly It will work regardless of the IDE.
I agree it will add some cognitive and maintenance burden, and it sounds like Firestore use `_` for other purpose as well, so I'm happy to disable this rule for Firestore if that's best for Firestore. Let me know :)

Reply:
Yeah, I think we want to disable it for Firestore.  Thanks / sorry!

Reply:
Done. disabled `no-unused-vars` for function params, but it still checks variables.